cloudflare/data_source_origin_ca_root_certificate.go

```diff
@@ -0,0 +1,48 @@
+package cloudflare
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func dataSourceCloudflareOriginCARootCertificate() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceCloudflareOriginCARootCertificateRead,
+
+		Schema: map[string]*schema.Schema{
+			"algorithm": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"rsa", "ecc"}, true),
+			},
+
+			"cert_pem": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceCloudflareOriginCARootCertificateRead(d *schema.ResourceData, meta interface{}) error {
+	algorithm := strings.ToLower(fmt.Sprintf("%v", d.Get("algorithm")))
```

Comment:
as we're using `strings.ToLower`, this should be explicit

```suggestion
	algorithm := strings.ToLower(fmt.Sprintf("%s", d.Get("algorithm")))
```

Reply:
Good catch